datafusion/expr/src/built_in_function.rs

```diff
@@ -949,12 +949,16 @@ impl BuiltinScalarFunction {
             }
             BuiltinScalarFunction::ArrayDims => Signature::any(1, self.volatility()),
             BuiltinScalarFunction::ArrayEmpty => Signature::any(1, self.volatility()),
-            BuiltinScalarFunction::ArrayElement => Signature::any(2, self.volatility()),
+            BuiltinScalarFunction::ArrayElement => {
+                Signature::array_and_element(self.volatility())
+            }
             BuiltinScalarFunction::ArrayExcept => Signature::any(2, self.volatility()),
             BuiltinScalarFunction::Flatten => Signature::any(1, self.volatility()),
             BuiltinScalarFunction::ArrayHasAll
             | BuiltinScalarFunction::ArrayHasAny
-            | BuiltinScalarFunction::ArrayHas => Signature::any(2, self.volatility()),
+            | BuiltinScalarFunction::ArrayHas => {
+                Signature::array_and_element(self.volatility())
```

Comment:
Can you add the example that works after this change.
I did not find one in array.slt

One of the example is `array_has(list, null)`

Reply:
`array_has_any` and `array_has_any` cannot use `array_and_element`, because their signature is array and array.